src/picture.cpp

```diff
@@ -154,6 +155,9 @@ namespace
 {
 image::locator::locator_finder_t locator_finder;
 
+// Use this for anything that may modify an image cache from a thread
```

Comment:
If the lock doesn't cover reading, how do we know that data will be valid while reader functions iterate through it?

I haven't looked in depth, but a first impression is that the mutex has been added to too few functions.

Reply:
Yes, it definitely has... the original guard was just on a single function :worried: .

So i'm kind of conflicted between doing it properly, and just leaving the mutex on that one function...

Reply:
On looking at it again i guess it does need to be done properly everywhere. That's pretty horrible.